web/src/SidebarResources.tsx

```diff
@@ -48,6 +49,72 @@ export function SidebarListSection(
   )
 }
 
+function SidebarItemsView(props: SidebarProps) {
+  return (
+    <>
+      {props.items.map((item) => (
+        <SidebarItemView
+          key={"sidebarItem-" + item.name}
+          item={item}
+          selected={props.selected === item.name}
+          pathBuilder={props.pathBuilder}
+          resourceView={props.resourceView}
+        />
+      ))}
+    </>
+  )
+}
+
+function SidebarLabelListSection(props: { label: string } & SidebarProps) {
+  if (props.items.length === 0) {
+    return null
+  }
+
+  return (
+    <>
+      <SidebarListSectionName>{props.label}</SidebarListSectionName>
+      <SidebarListSectionItems>
+        <SidebarItemsView {...props} />
+      </SidebarListSectionItems>
+    </>
+  )
+}
+
+function SidebarGroupedByLabels(props: SidebarProps) {
+  const labelsToResources: { [key: string]: SidebarItem[] } = {}
+  const unlabeled: SidebarItem[] = []
+
+  props.items.forEach((item) => {
+    if (item.labels.length) {
+      item.labels.forEach((label) => {
+        if (!labelsToResources.hasOwnProperty(label)) {
+          labelsToResources[label] = []
+        }
+
+        labelsToResources[label].push(item)
+      })
+    } else {
+      unlabeled.push(item)
+    }
+  })
+
+  const labels = Object.keys(labelsToResources)
+
+  return (
+    <>
+      {labels.map((label) => (
+        <SidebarLabelListSection
+          {...props}
+          key={`sidebarItem-${label}`}
+          label={label}
+          items={labelsToResources[label]}
+        />
+      ))}
+      <SidebarLabelListSection {...props} label="unlabeled" items={unlabeled} />
```

Comment:
probably more of a question for Surbhi, but I feel like it's going to be confusing to have both an "unlabeled" section and "uncategorized" resource

Reply:
ya, that's a good question! @SurbhiSGupta may have user stories/feedback that she can share. right now, i just have the `unlabeled` section implemented.